server/server.go

```diff
@@ -1582,6 +1583,12 @@ func (s *Server) Start() {
 			}
 			return true
 		})
+
+		// We may be in a mixed mode here.
+		if !s.standAloneMode() && s.globalAccountOnly() {
```

Comment:
This can affect lots of test because as soon as you use DefaultOptions() which set a cluster port, then this test will be true: that is, stanAloneMode() is false because cluster port is defined. And if test does not specify specific accounts, then we add JS subscriptions.. maybe it is correct, but seem to be a bit intrusive..

Reply:
I will check to make sure its not processed twice from above. Will also rebase.

Reply:
Without that change you can have a case where some $G are enabled and those on servers that do not have JS enabled will not properly allow things to be forwarded.

I could take a different approach now with clustering and not turn anything on at all.. WDYT?

Reply:
Not sure what the approach would be, but I hope you get my point that in a situation where I have simply clustering enabled (as in routes) and no JS at all, it seem a bit intrusive to have the system add things related to JetStream. Now it could be that there is something JS related in this test or something that has to do with default options, etc..
Now if we want to support mixed-mode without the user configuring anything in the non-js node, then I understand that the server has to do some automatic configuration...

Reply:
This was a misconfiguration, will remove all of this code.